internal/http/services/owncloud/ocdav/tpc.go

```diff
@@ -0,0 +1,427 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package ocdav
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	gateway "github.com/cs3org/go-cs3apis/cs3/gateway/v1beta1"
+	rpc "github.com/cs3org/go-cs3apis/cs3/rpc/v1beta1"
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	typespb "github.com/cs3org/go-cs3apis/cs3/types/v1beta1"
+	"github.com/cs3org/reva/internal/http/services/datagateway"
+	"github.com/cs3org/reva/pkg/appctx"
+	"github.com/cs3org/reva/pkg/rhttp"
+)
+
+const (
+	// PerfMarkerResponseTime corresponds to the interval at which a performance marker is sent back to the TPC client
+	PerfMarkerResponseTime float64 = 5
+)
+
+// PerfResponse provides a single chunk of permormance marker response
+type PerfResponse struct {
+	Timestamp time.Time
+	Bytes     uint64
+	Index     int
+	Count     int
+}
+
+func (p *PerfResponse) getPerfResponseString() string {
+	var response string
+	response += "Perf Marker\n"
+	response += "Timestamp: " + strconv.FormatInt(p.Timestamp.Unix(), 10) + "\n"
+	response += "Stripe Bytes Transferred: " + strconv.FormatUint(p.Bytes, 10) + "\n"
+	response += "Strip Index: " + strconv.Itoa(p.Index) + "\n"
+	response += "Total Stripe Count: " + strconv.Itoa(p.Count) + "\n"
+	response += "End\n"
+	return response
+}
+
+// WriteCounter counts the number of bytes transferred and reports
+// back to the TPC client about the progress of the transfer
+// through the performance marker response stream.
+type WriteCounter struct {
+	Total    uint64
+	PrevTime time.Time
+	w        http.ResponseWriter
+}
+
+// SendPerfMarker flushes a single chunk (performance marker) as
+// part of the chunked transfer encoding scheme.
+func (wc *WriteCounter) SendPerfMarker(size uint64) {
+	flusher, ok := wc.w.(http.Flusher)
+	if !ok {
+		panic("expected http.ResponseWriter to be an http.Flusher")
+	}
+	perfResp := PerfResponse{time.Now(), size, 0, 1}
+	pString := perfResp.getPerfResponseString()
+	fmt.Fprintln(wc.w, pString)
+	flusher.Flush()
+}
+
+func (wc *WriteCounter) Write(p []byte) (int, error) {
+
+	n := len(p)
+	wc.Total += uint64(n)
+	NowTime := time.Now()
+
+	diff := NowTime.Sub(wc.PrevTime).Seconds()
+	if diff >= PerfMarkerResponseTime {
+		wc.SendPerfMarker(wc.Total)
+		wc.PrevTime = NowTime
+	}
+	return n, nil
+}
+
+// Uncomment to use for debugging
```

Comment:
I wouldn't leave this here. You could use the logger, and log this on debug level, so when developing against you just need to raise the logging level